File: BOJ/#1193 분수찾기/분수찾기.cpp
#include <iostream>
using namespace std;

int main() {
    int X; cin>>X;

    int cpyX=X, cnt=0;
    while(cpyX > 0) {
        cnt++;
        cpyX -= cnt;
    }
    cpyX += cnt;

    // cnt가 홀수면 오른쪽 위 대각선
    if(cnt%2) {
        int up = cnt-(cpyX-1);
        int down = cpyX;
        printf("%d/%d\n", up, down);
    }
    // cnt가 짝수면 왼쪽 아래 대각선
    else {
        int up = cpyX;
        int down = cnt-(cpyX-1);
        printf("%d/%d\n", up, down);
    }
}

File: BOJ/#1193 분수찾기/분수찾기.py
X = int(input())

cpyX,cnt = X,0
while cpyX > 0:
    cnt += 1
    cpyX -= cnt
cpyX += cnt

if cnt%2 == 1: print("%d/%d"%(cnt-(cpyX-1), cpyX))
else: print("%d/%d"%(cpyX, cnt-(cpyX-1)))
